src/librustc_driver/lib.rs

```diff
@@ -360,21 +361,71 @@ fn handle_explain(code: &str,
     };
     match descriptions.find_description(&normalised) {
         Some(ref description) => {
-            // Slice off the leading newline and print.
-            print!("{}", &(&description[1..]).split("\n").map(|x| {
+            let output = &(&description[1..]).split("\n").map(|x| {
                 format!("{}\n", if x.starts_with("```") {
                     "```"
                 } else {
                     x
                 })
-            }).collect::<String>());
+            }).collect::<String>();
+
+            show_content_with_pager(output);
+
         }
         None => {
             early_error(output, &format!("no extended information for {}", code));
         }
     }
 }
 
+#[cfg(not(windows))]
+fn spawn_pager_with_file(file_path: &PathBuf) -> io::Result<Output> {
+    use std::os::unix::io::FromRawFd;
+    let raw_file_descriptor_id = 2; // stdout
+
+    let pager = env::var("PAGER").unwrap_or(String::from("less"));
+
+    unsafe {
+        Command::new(pager)
+            .arg(file_path)
+            .stdout(Stdio::from_raw_fd(raw_file_descriptor_id))
+            .output()
+    }
+}
+
+#[cfg(windows)]
+fn spawn_pager_with_file(file_path: &PathBuf) -> io::Result<Output> {
+    let pager = env::var("PAGER").unwrap_or(String::from("more.com"));
+
+    Command::new(pager)
+        .arg(file_path)
+        .output()
+}
+
+fn show_content_with_pager(content: &str) {
+
+    let file_name = "rustc_pager_content";
+    let mut file_path = env::temp_dir();
+    file_path.push(file_name);
+    let temp_file = File::create(&file_path);
+
+    if temp_file.is_err() {
+        println!("unable to create temporary file for pager content");
+        return
+    }
+
+    temp_file.unwrap()
+        .write_all(content.as_bytes())
+        .expect("unable to write pager content to temporary file");
+
+    if spawn_pager_with_file(&file_path).is_err() {
+        println!("unable to spawn pager process");
```

Comment:
hm, I think ideally we'd send this to stderr and (as in all other cases in this function) send the output to stdout. We shouldn't ever *not* show the output because a pager is missing.